File: Cursor/database/init_database.py
#!/usr/bin/env python3

import sqlite3
import os
from pathlib import Path

db_dir = Path(__file__).parent
db_path = db_dir / 'team_directory.db'

sample_employees = [
    ('Ahmed', 'Khan', 'Software Engineer'),
    ('Fatima', 'Ali', 'Product Manager'),
    ('Hassan', 'Malik', 'UX Designer'),
    ('Ayesha', 'Raza', 'Data Analyst'),
    ('Usman', 'Sheikh', 'DevOps Engineer'),
    ('Zainab', 'Iqbal', 'Frontend Developer'),
    ('Bilal', 'Hussain', 'Backend Developer'),
    ('Sana', 'Ahmed', 'QA Engineer')
]

try:
    # Connect to database (creates file if it doesn't exist)
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    print(f'Connected to SQLite database: {db_path}')
    
    # Create table
    cursor.execute('''
        CREATE TABLE IF NOT EXISTS Employees (
            ID INTEGER PRIMARY KEY AUTOINCREMENT,
            FirstName TEXT NOT NULL,
            LastName TEXT NOT NULL,
            Role TEXT NOT NULL
        )
    ''')
    print('Employees table created/verified.')
    
    # Clear existing data (optional)
    cursor.execute('DELETE FROM Employees')
    print('Table cleared.')
    
    # Insert sample data
    cursor.executemany('''
        INSERT INTO Employees (FirstName, LastName, Role)
        VALUES (?, ?, ?)
    ''', sample_employees)
    
    print(f'Inserted {len(sample_employees)} employee records.')
    
    # Verify data
    cursor.execute('SELECT * FROM Employees ORDER BY LastName, FirstName')
    rows = cursor.fetchall()
    
    print('\nDatabase contents:')
    print('ID | First Name | Last Name  | Role')
    print('-' * 60)
    for row in rows:
        print(f'{row[0]:2} | {row[1]:10} | {row[2]:10} | {row[3]}')
    
    # Commit changes
    conn.commit()
    print('\nDatabase initialized successfully!')
    
except sqlite3.Error as e:
    print(f'Database error: {e}')
except Exception as e:
    print(f'Error: {e}')
finally:
    if conn:
        conn.close()
        print(f'Database connection closed.')

File: database/init_database.js
import sqlite3 from 'sqlite3';
import path from 'path';
import fs from 'fs';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const dbPath = path.join(__dirname, 'team_directory.db');

const sampleEmployees = [
    { FirstName: 'Ahmed', LastName: 'Khan', Role: 'Software Engineer' },
    { FirstName: 'Fatima', LastName: 'Ali', Role: 'Product Manager' },
    { FirstName: 'Hassan', LastName: 'Malik', Role: 'UX Designer' },
    { FirstName: 'Ayesha', LastName: 'Raza', Role: 'Data Analyst' },
    { FirstName: 'Usman', LastName: 'Sheikh', Role: 'DevOps Engineer' },
    { FirstName: 'Zainab', LastName: 'Iqbal', Role: 'Frontend Developer' },
    { FirstName: 'Bilal', LastName: 'Hussain', Role: 'Backend Developer' },
    { FirstName: 'Sana', LastName: 'Ahmed', Role: 'QA Engineer' }
];

// Create or open database
const db = new sqlite3.Database(dbPath, (err) => {
    if (err) {
        console.error('Error opening database:', err.message);
        process.exit(1);
    }
    console.log('Connected to SQLite database.');
});

// Create table
db.serialize(() => {
    db.run(`
        CREATE TABLE IF NOT EXISTS Employees (
            ID INTEGER PRIMARY KEY AUTOINCREMENT,
            FirstName TEXT NOT NULL,
            LastName TEXT NOT NULL,
            Role TEXT NOT NULL
        )
    `, (err) => {
        if (err) {
            console.error('Error creating table:', err.message);
        } else {
            console.log('Employees table created/verified.');
        }
    });

    // Clear existing data (optional)
    db.run('DELETE FROM Employees', (err) => {
        if (err) {
            console.error('Error clearing table:', err.message);
        } else {
            console.log('Table cleared.');
        }
    });

    // Insert sample data
    const stmt = db.prepare('INSERT INTO Employees (FirstName, LastName, Role) VALUES (?, ?, ?)');
    
    sampleEmployees.forEach((employee) => {
        stmt.run(employee.FirstName, employee.LastName, employee.Role, (err) => {
            if (err) {
                console.error('Error inserting employee:', err.message);
            }
        });
    });

    stmt.finalize((err) => {
        if (err) {
            console.error('Error finalizing statement:', err.message);
        } else {
            console.log(`Inserted ${sampleEmployees.length} employee records.`);
        }

        // Verify data
        db.all('SELECT * FROM Employees', (err, rows) => {
            if (err) {
                console.error('Error querying data:', err.message);
            } else {
                console.log('\nDatabase contents:');
                console.table(rows);
            }

            // Close database
            db.close((err) => {
                if (err) {
                    console.error('Error closing database:', err.message);
                } else {
                    console.log('\nDatabase initialized successfully!');
                    console.log(`Database file: ${dbPath}`);
                }
            });
        });
    });
});
